code/game/mecha/mech_prosthetics.dm

```diff
@@ -172,46 +172,32 @@
 				qdel(I)
 		return
 
-	var/material
-	switch(I.type)
-		if(/obj/item/stack/material/gold)
-			material = "gold"
-		if(/obj/item/stack/material/silver)
-			material = "silver"
-		if(/obj/item/stack/material/diamond)
-			material = "diamond"
-		if(/obj/item/stack/material/phoron)
-			material = "phoron"
-		if(/obj/item/stack/material/steel)
-			material = DEFAULT_WALL_MATERIAL
-		if(/obj/item/stack/material/glass)
-			material = "glass"
-		if(/obj/item/stack/material/uranium)
-			material = "uranium"
-		if(/obj/item/stack/material/plasteel)
-			material = "plasteel"
+	if(istype(I,/obj/item/stack/material))
+		var/obj/item/stack/material/S = I
+		if(!S.material in materials)
```

Comment:
It's sort of outside the scope of this project to rewrite how other machines work, to be honest. Your not-this catch is good though so I'll change that real quick.